doc/safety.md

```diff
@@ -0,0 +1,58 @@
+# SECURITY
+
+This document describes the security recommendations and issues that can be helpful for users of Bitcoin Core.
+
+ℹ️ There is no patch for human stupidity.
+
+Download Bitcoin Core only from below links, verify the integrity of the download before using:
+
+https://bitcoincore.org/en/download/
+
+https://github.com/bitcoin/bitcoin/releases
+
+Downloading Bitcoin Core from untrusted sources can result in using a [malware](https://bitcoin.stackexchange.com/a/107738/) that can steal your bitcoin and compromise your computer.
+
+### P2P
+
+Using non-default ports and non-listening nodes can improve security.
+
+DNS seeds are used for bootstrapping the P2P network. If one of the domains used for default DNS seeds is hacked, this can impact nodes. A node checks 3 seeds at a time, so it would require coordination for a malicious seed to affect the network. You can use own DNS seed for bootstrapping.
```

Comment:
> Nobody has ever run their own DNS seed in order to bootstrap a synching node, the suggestion is complete nonsense.

I have not written history in this sentence. Anyone CAN run DNS seed using https://github.com/sipa/bitcoin-seeder

Reply:
Just because you could doesn't mean anybody has, or should. There is no reasonable person who would suggest a user run their own DNS seed, a system that requires large resources and setup cost, in order to synd their own node.

Reply:
It also makes no sense, because it just shifts the problem of bootstrapping to the seeder code without improving the solution.
I don't think the seeder does anything else than ask other dns seeds to bootstrap itself ([code](https://github.com/sipa/bitcoin-seeder/blob/33d0400d0c1026de84f29c32716849ee4a86828e/main.cpp#L421-L444)). If all other DNS seeds feed you malicious IPs, you have gained nothing by running your own DNS seed as opposed to just running a node.

Reply:
There are two things shared in doc. 1. Running own DNS seed. It makes sense because you won't trust default DNS seeds. 2. Seed node can be used for bootstrapping. Difference is explained in the link.

Reply:
> 1. Running own DNS seed. It makes sense because you won't trust default DNS seeds.

Disagree. I don't think the sentence  "You can use own DNS seed for bootstrapping." is helpful, because even if I ran a DNS seed using https://github.com/sipa/bitcoin-seeder as suggested, this DNS seed would also need to bootstrap somehow to get going initially. 

As indicated by the link in my previous post, the bootstrapping mechanism implemented in the DNS seeder not only queries other existing DNS seeds similar to bitcoin core -  so you still "trust default DNS seeds" - but is otherwise inferior to the one used by core (no fixed seeds as fallback, outdated hardcoded DNS seed list) - which makes sense because bitcoin-seeder is just not meant to be used for the use case you are suggesting here.

Reply:
Running your own DNS seed makes no sense, for multiple reasons:
* Seeds are just a way to find an initial connection to the network. If you don't believe the DNS seeds hardcoded in Bitcoin Core are trustworthy enough for that, you should manually make connections to nodes you do trust (using `-seednode` / `-addnode`). You can also disable the DNS seeds (`-dnsseed=0`) in that case.
* Bitcoin Core offers no functionality to using your own DNS seed.

Reply:
>Disagree. I don't think the sentence "You can use own DNS seed for bootstrapping." is helpful, because even if I ran a DNS seed using https://github.com/sipa/bitcoin-seeder as suggested, this DNS seed would also need to bootstrap somehow to get going initially.

@mzumsande 

Yes. Sentence can be improved and I am going to add/update a few things but did not because wanted to spend less time doing pull requests on weekend. Looks like this PR has lot of interest from reviewers so I will do it in next few hours.

The attack explained in doc assumes most of the DNS seeds are safe but they might get compromised at some point. I am not sure about the code you shared @sipa can explain that better as some domains dont even resolve to anything in that. Example: bitseed.xf2.org However even if we assume these are used once in a lifetime it won't be an issue using 1-2 or get IPs based on your own crawlers.


> Running your own DNS seed makes no sense, for multiple reasons:

> Seeds are just a way to find an initial connection to the network. If you don't believe the DNS seeds hardcoded in Bitcoin Core are trustworthy enough for that, you should manually make connections to nodes you do trust (using -seednode / -addnode). You can also disable the DNS seeds (-dnsseed=0) in that case.

> Bitcoin Core offers no functionality to using your own DNS seed.

@sipa `-seednode` thing is explained in the link and mentioned one word but I can rephrase it in a better way. Will add `-dnsseed=0` on next update. 

Bitcoin Core does not offer functionality to use own DNS seed but there has to be one paragraph about trust assumption even if I remove the part that suggested running own DNS seed which is technically possible as there can be several ways to bootstrap.

Reply:
Yes, I definitely agree with explaining the trust assumptions around DNS seeds, and that one may want to avoid them. But "running your own DNS seed" is pointless as a solution for it.

Reply:
It is removed